# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1472221264 -7200
#      Fri Aug 26 16:21:04 2016 +0200
# Node ID c5da5e5e984613e2e5574b942e95fd5a5b6ab0e3
# Parent  6cbbd550aa919df26631dc5dec91d2e2c00c6ec7
Fix inconsistent on-screen/printout patch and surface edges (bug #48873).

* gl-render.cc (opengl_renderer::draw_surface): Don't draw lines at all if
linestyle is "none".
* gl-render.cc (opengl_renderer::draw_patch): Ditto.

diff --git a/libinterp/corefcn/gl-render.cc b/libinterp/corefcn/gl-render.cc
--- a/libinterp/corefcn/gl-render.cc
+++ b/libinterp/corefcn/gl-render.cc
@@ -2327,17 +2327,17 @@ namespace octave
               glDisable (GL_LIGHTING);
           }
         else
           {
             // FIXME: implement transparency
           }
       }
 
-    if (! props.edgecolor_is ("none"))
+    if (! props.edgecolor_is ("none") && ! props.linestyle_is ("none"))
       {
         if (props.get_edgealpha_double () == 1)
           {
             if (ec_mode == UNIFORM)
               {
                 glColor3dv (ecolor.data ());
                 if (fl_mode > 0)
                   {
@@ -2907,17 +2907,17 @@ namespace octave
               glDisable (GL_LIGHTING);
           }
         else
           {
             // FIXME: implement transparency
           }
       }
 
-    if (! props.edgecolor_is ("none"))
+    if (! props.edgecolor_is ("none") && ! props.linestyle_is ("none"))
       {
         // FIXME: adapt to double-radio property
         if (props.get_edgealpha_double () == 1)
           {
             if (ec_mode == UNIFORM)
               {
                 glColor3dv (ecolor.data ());
                 if (el_mode > 0)
